### PR TITLE
Ripple Effect for navigation Icon in v2 App Bar

### PR DESCRIPTION
**Pull request by Anubhvv**
### Problem 
The user waned to show ripple effect in Navigation Icon.
### Root cause 
The modifier wasn't extended
### Fix

- The modifier "NavigationIconModifier" can be passed to change default behaviour/ look.
- Default behaviour of ripple on navigation icon click has been added

### Validations

(how the change was tested, including both manual and automated tests)

### Screenshots

Before: (No ripple)

https://github.com/microsoft/fluentui-android/assets/68989156/93dc1c1d-e80f-4507-b004-1a5902c3448f

After (Ripple Effect)

https://github.com/microsoft/fluentui-android/assets/68989156/d1ad1264-c040-4d6e-b57b-61e7e1120e98

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] Automated Tests
- [ ] Documentation and demo app examples
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and RTL layouts
- [ ] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: fluentui_topappbars/src/main/java/com/microsoft/fluentui/tokenized/AppBar.kt [hunk 1]
```diff
@@ -140,7 +141,9 @@ fun AppBar(
                 if (appBarSize != AppBarSize.Large && navigationIcon.isIconAvailable()) {
                     Icon(
                         navigationIcon,
-                        modifier = Modifier
+                        modifier =
+                        Modifier
+                            .then(navigationIconModifier)
```
Comment (mishramayank1) [inline fluentui_topappbars/src/main/java/com/microsoft/fluentui/tokenized/AppBar.kt:146]: I believe the ripple should be shown by default. Please check if that's not the case & need to adapt as per user scenario

Comment (Anubhvv) [reply fluentui_topappbars/src/main/java/com/microsoft/fluentui/tokenized/AppBar.kt]: Hey, I have added default ripple behavior as you suggested, and also provided this modifier incase user wants to override ripple color, radius, etc.

File: fluentui_topappbars/src/main/java/com/microsoft/fluentui/tokenized/AppBar.kt [hunk 2]
```diff
@@ -79,6 +83,7 @@ fun AppBar(
     logo: @Composable (() -> Unit)? = null,
     searchMode: Boolean = false,
     navigationIcon: FluentIcon = FluentIcon(AppBarIcons.Arrowback, flipOnRtl = true),
+    navigationIconModifier: Modifier? = null,
```
Comment (mishramayank1) [inline fluentui_topappbars/src/main/java/com/microsoft/fluentui/tokenized/AppBar.kt:86]: need not to expose navigationIconModifier

**Comment by mishramayank1** [comment]
Also test in dark mode

File: fluentui_topappbars/src/main/java/com/microsoft/fluentui/tokenized/AppBar.kt (hunk 2 above)
Comment (Anubhvv) [reply fluentui_topappbars/src/main/java/com/microsoft/fluentui/tokenized/AppBar.kt]: okay, removed

**Review by mishramayank1 — APPROVED** [review]

**Pull request merged**